Move match preference string to enum conversion to MatchPreference

jzwang8 opened the pull request:
Files changed:
- MatchPreference.java: add forStringValue for conversion from string to enum
- AddParticipantServlet.java: remove conversion method

kmh287 commented on backend/src/main/java/com/google/sps/data/MatchPreference.java:
```diff
@@ -41,4 +41,17 @@ public static MatchPreference forIntValue(int value) {
     throw new IllegalStateException("Unknown enum value.");
   }
+
+  public static MatchPreference forStringValue(String value) {
+    switch (value) {
+      case "different":
+        return MatchPreference.DIFFERENT;
+      case "any":
+        return MatchPreference.ANY;
+      case "similar":
+        return MatchPreference.SIMILAR;
+      default:
+        return null;
+    }
```
There's a good opportunity for a test here. How should this method handle "ANY" or "SiMiLaR"?

jzwang8 replied:
I added toLowerCase() to value, but also value should always be one of those three strings since that's what the frontend passes, so if it's none of the above, then an error should be thrown

kmh287 replied:
Looks good. There's still an opportunity here for a (short) unit test.

jzwang8 replied:
Added tests

tcwanggoogle commented on backend/src/main/java/com/google/sps/data/MatchPreference.java:
```diff
@@ -41,4 +41,17 @@ public static MatchPreference forIntValue(int value) {
+      case "similar":
+        return MatchPreference.SIMILAR;
+      default:
+        return null;
```
Should we throw exception here like the forIntValue()? Otherwise, anywhere you want to use MatchPreference, you have to check for null.

kmh287 commented:
Nice job on the tests

jzwang8 commented:
> LGTM! One general comment, since we have added a UsernameService class that now implements the getUsername functionality, WDYT about taking the functionality out of this class? I think we can wait and do it in a follow up PR so it's consistent in all servlets if that works better.

Yep, I created this PR before we refactored servlets/tests. Other servlet PRs have the UsernameService class since that's not in master yet and they all depend on that.
